src/components/QueryOutput/QueryOutput.js

```diff
@@ -0,0 +1,1988 @@
+      })
+
+      const uniqueYears = [...allYears]
+        .filter(onlyUnique)
+        .sort()
+        .reduce((map, title, i) => {
+          map[title] = i + 1
+          return map
+        }, {})
+
+      // Generate new column array
+      const pivotTableColumns = [
+        {
+          ...this.tableColumns[dateColumnIndex],
+          title: 'Month',
+          name: 'Month',
+          field: '0',
+          frozen: true,
+          visible: true,
+          is_visible: true,
+          type: 'DATE_STRING',
+          datePivot: true,
+          origColumn: columns[dateColumnIndex],
+          pivot: true,
+          cssClass: 'pivot-category',
+          sorter: dateSortFn,
+          headerFilter: false,
+        },
+      ]
+
+      Object.keys(uniqueYears).forEach((year, i) => {
+        pivotTableColumns.push({
+          ...this.tableColumns[numberColumnIndex],
+          origColumn: this.tableColumns[numberColumnIndex],
+          origValues: {},
+          name: year,
+          title: year,
+          field: `${i + 1}`,
+          visible: true,
+          is_visible: true,
+          headerFilter: false,
+        })
+      })
+
+      const pivotTableData = makeEmptyArray(Object.keys(uniqueYears).length, 12)
+      const pivotOriginalColumnData = {}
+
+      // Populate first column
+      MONTH_NAMES.forEach((month, i) => {
+        pivotTableData[i][0] = month
+      })
+
+      // Populate remaining columns
+      tableData.forEach((row) => {
+        const year = this.formatDatePivotYear(row, dateColumnIndex)
+        const month = this.formatDatePivotMonth(row, dateColumnIndex)
+
+        const yearNumber = uniqueYears[year]
+        const monthNumber = MONTH_NAMES.findIndex((m) => month === m)
+
+        const pivotColumnIndex = pivotTableColumns.findIndex(
+          (col) => col.name === year
+        )
+
+        if (monthNumber >= 0 && yearNumber) {
+          pivotTableData[monthNumber][yearNumber] = row[numberColumnIndex]
+          pivotOriginalColumnData[year] = {
+            ...pivotOriginalColumnData[year],
+            [month]: row[dateColumnIndex],
+          }
+          pivotTableColumns[pivotColumnIndex].origValues[month] = {
+            name: this.tableColumns[dateColumnIndex]?.name,
+            value: row[dateColumnIndex],
+          }
+        }
+      })
+
+      this.pivotOriginalColumnData = pivotOriginalColumnData
+      this.pivotTableColumns = pivotTableColumns
+      this.pivotTableData = pivotTableData
+      this.numberOfPivotTableRows = 12
+      this.setPivotTableConfig()
+    } catch (error) {
+      console.error(error)
+      const supportedDisplayTypes = this.getCurrentSupportedDisplayTypes()
+      this.setState({
+        displayType: 'table',
+        supportedDisplayTypes: supportedDisplayTypes.filter(
+          (displayType) => displayType !== 'pivot_table'
+        ),
+      })
+    }
+  }
+
+  generatePivotTableData = ({ isFirstGeneration } = {}) => {
+    try {
```

Comment:
CAn you explain why you used 'let' instead of 'const' for tableData, uniqueValues0, uniqueValues1 etc?